### PR TITLE
RFC: Julia

### PR DESCRIPTION
**Pull request by iamed2**
I'd like to add Julia to the languages but it doesn't seem to be a supported apt package on Travis yet. Are you interested in having Julia here? If so, should I try to get Travis supporting Julia's apt package or should I download binaries myself in the build script?

Also, have I done everything necessary to add Julia support otherwise?

### CODE REVIEW

**Comment by iamed2** [comment]
Actually it looks like julia isn't available in Ubuntu Precise even outside of travis so downloading binaries is the only option.

**Comment by shepmaster** [comment]
This is great!

As a heads-up, I'm using the container-based infrastructure on Travis. That means that you can't do any sudo-like stuff. Will you be able to download and install the Julia binaries as a plain-old-user? I'm also going to check with my Travis contacts and see if there's something nicer we can do.

**Comment by shepmaster** [comment]
Ah, it looks like the [official Julia environment](https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/script/julia.rb#L38-L49) does much the same - it downloads Julia, unpacks to $HOME, and adds it to the path.

**Comment by iamed2** [comment]
Looks like this is good to go now!

**Comment by shepmaster** [comment]
This is super exciting, thanks for the hard work! :heart: 

How comfortable do you feel with git? I'd prefer if you could clean up your commits a bit. Your commit titles have a reasonable breakdown (code / HTML / Travis), but your "documentation" commit contains some code — I'd like that to be part of the "code" commit. I'm also OK if you squash the "finalizing immutables" commit into the "code" commit.

If you don't feel comfortable doing that in git, I'm more than happy to do my own pedantic moving around, so don't feel obligated!

File: .travis.yml [hunk 1]
```diff
@@ -16,4 +19,5 @@ script:
...
+    - JULIA_URL='https://julialang.s3.amazonaws.com/bin/linux/x64/0.4/julia-0.4.1-linux-x86_64.tar.gz'
+    - secure: tQbG88b4WOHHQ1VK58Ah4I/XYEW0AiFcmcnVvARizSLlJk6bVx0S9neNvEi9PxTihSfidsC6TGK+gGRpsPuIJjKXs7koKMLtS0DoHjhpU72RX+QXIgvu2Yr01ZsMqpGOPhdnN4raWqDQXUqRLTJ/7xMgA81O+rkeYzZlKF21Z536IoEHT65PkxJdBh9kEP7VsAcXwBO4lGfJxLm2sBJaRf07OvlwxBNF5kSr4T0y+oYRes658cxAhiRXl316WBlFSNm4g4lSB9bewlXDYEL1Z+r0abhJIyQxN+2xvrDW9zdjGXBRX8pjTfALTHpx3hmlk24EoLQaiPQyNC3p1bvNnpglLSgy2WEFtrzQR05okyNoNn0BaemZHvCKjWsi4btYjmOKXzSZ8OMqyHAl1WNxwv5NLixJI38h8N+M1sqRX64WwZZx0gCWAKjMkh3Kaa4CNMvkP4M+4y2IalxnwAaDtZbQLkthAP4xr+rnsXOEwPcxKUBMdUuMuIYdH5QjC2kj3jOMTq2UjcpvrOA+2JSAvsYDOlj0fUrjA2YaHb35yacyNnGA/FjQ79PRtkR0FGzYISTuRJBX6CIVu99WF0yjFh3iKZOTxngbkrRqdf+DuCu/IBmSoGfHmDCfSKOxCovdpJD9kvuWpBf92nlnY+B4UZXeDneOO6Jg+05eB312NXo=
```
Comment (shepmaster) [inline .travis.yml]: Any idea why this line shows as a diff? I'd prefer if it didn't, just as it is the secure bits.

File: site/tuples/index.md [hunk 2]
```diff
@@ -70,3 +70,13 @@ To mirror the tuple structure definition, we create a `struct` using
 the `StructLayout` property and define the layout as sequential. We
 also provide some implicit conversion operators to make going between
 types fairly seamless.
+
+## Julia
+
+{% example src/main.jl %}
+
+To mirror the tuple structure definition, we create a type (note: we
+could also have used an immutable type with no other changes) with the
```
Comment (shepmaster) [inline site/tuples/index.md]: Do you think this is still true, given that you had to change the object test from `immutable` to `type`?

File: site/tuples/index.md [hunk 3]
```diff
@@ -70,3 +70,13 @@ To mirror the tuple structure definition, we create a `struct` using
...
+same size and order of fields as the rust definition. `ccall` allows
```
Comment (shepmaster) [inline site/tuples/index.md]: Capitalize Rust as it's a proper noun.

**Comment by shepmaster** [comment]
I'd still love to have Julia support, but since this has been idle for a while, I'm going to close it for now. Feel free to reopen when interest returns! 🌴

**Pull request closed without merging**